js/ui/date_box/ui.date_box.base.js

```diff
@@ -309,6 +309,7 @@ var DateBox = DropDownEditor.inherit({
 
     _init: function() {
         this._initStrategy();
+        this.$element().addClass("dx-show-invalid-badge");
```

Comment:
Replace class name with a const. It should be defined at the head of file like `DX_AUTO_WIDTH_CLASS`